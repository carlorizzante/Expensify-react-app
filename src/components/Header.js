import React from 'react';
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import { startLogout } from '../actions/auth.js';

export const Header = (props) => (
  <header>
    <h1>Expensify App</h1>
    <nav>
      <ul>
        <li><NavLink to="/dashboard" activeClassName="is-active" exact>Dashboard</NavLink></li>
        <li><NavLink to="/create"    activeClassName="is-active" exact>Create Expense</NavLink></li>
        <li><button
          className="btn btn-primary"
          onClick={ props.startLogout }
        >Logout</button></li>
      </ul>
    </nav>
  </header>
);

const mapDispatchToProps = (dispatch) => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(null, mapDispatchToProps)(Header);
