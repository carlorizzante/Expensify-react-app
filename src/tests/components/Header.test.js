import React from 'react';
// import ReactShallowRenderer from 'react-test-renderer/shallow';
import { shallow } from 'enzyme';
// import toJson from 'enzyme-to-json'; // added via jest.config.json

import { Header } from '../../components/Header.js';

// // Testing with react-test-renderer
// // This test is redundant, ReactShallowRenderer is substituted by Enzyme
// test('should render Header component correctly', () => {
//   const renderer = new ReactShallowRenderer();
//   renderer.render(<Header/>);
//   expect(renderer.getRenderOutput()).toMatchSnapshot();
// });

// Testing with Enzyme
test('should render Header component correctly', () => {
  const wrapper = shallow(<Header startLogout={ () => {} }/>);
  expect(wrapper.find('h1').length).toBe(1);
  expect(wrapper.find('h1').text()).toBe("Expensify App");

  expect(wrapper.find('nav').length).toBe(1);
  expect(wrapper.find('ul').length).toBe(1);
  expect(wrapper.find('li').length).toBeGreaterThanOrEqual(1);

  expect(wrapper).toMatchSnapshot();
  // expect(toJson(wrapper)).toMatchSnapshot(); // added via jest.config.json
});

test('should call startLogout on button click', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={ startLogout }/>);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
