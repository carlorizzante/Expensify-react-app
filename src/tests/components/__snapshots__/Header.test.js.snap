// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Header component correctly 1`] = `
<header>
  <h1>
    Expensify App
  </h1>
  <nav>
    <ul>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          exact={true}
          to="/dashboard"
        >
          Dashboard
        </NavLink>
      </li>
      <li>
        <NavLink
          activeClassName="is-active"
          aria-current="page"
          exact={true}
          to="/create"
        >
          Create Expense
        </NavLink>
      </li>
      <li>
        <button
          className="btn btn-primary"
          onClick={[Function]}
        >
          Logout
        </button>
      </li>
    </ul>
  </nav>
</header>
`;
